modules/quotas-requiring-explicit-quota.adoc

```diff
@@ -4,48 +4,131 @@
+
+* Install the OpenShift CLI (`oc`).
 
 .Procedure
 
-To require explicit quota to consume a particular resource:
+. Add a resource quota definition to a project request template:
++
+* If a project request template does not exist in a cluster:
+.. Create a bootstrap project template and output it to a file called `template.yaml`:
++
+----
+$ oc adm create-bootstrap-project-template -o yaml > template.yaml
+----
++
+.. Add a resource quota definition to `template.yaml`. The following example defines a resource quota named 'storage-consumption'. The definition must be added before the `parameters:` section in the template:
++
+[source,yaml]
+----
+- apiVersion: v1
+  kind: ResourceQuota
+  metadata:
+    name: storage-consumption
+  spec:
+    hard:
+      persistentvolumeclaims: "10" <1>
+      requests.storage: "50Gi" <2>
+      gold.storageclass.storage.k8s.io/requests.storage: "10Gi" <3>
+      silver.storageclass.storage.k8s.io/requests.storage: "20Gi" <4>
+      silver.storageclass.storage.k8s.io/persistentvolumeclaims: "5" <5>
+      bronze.storageclass.storage.k8s.io/requests.storage: "0" <6>
+      bronze.storageclass.storage.k8s.io/persistentvolumeclaims: "0" <7>
+----
+<1> The total number of persistent volume claims in a project.
+<2> Across all persistent volume claims in a project, the sum of storage requested cannot exceed this value.
+<3> Across all persistent volume claims in a project, the sum of storage requested in the gold storage class cannot exceed this value.
+<4> Across all persistent volume claims in a project, the sum of storage requested in the silver storage class cannot exceed this value.
+<5> Across all persistent volume claims in a project, the total number of claims in the silver storage class cannot exceed this value.
+<6> Across all persistent volume claims in a project, the sum of storage requested in the bronze storage class cannot exceed this value. When this value is set to `0`, the bronze storage class cannot request storage.
+<7> Across all persistent volume claims in a project, the sum of storage requested in the bronze storage class cannot exceed this value. When this value is set to `0`, the bronze storage class cannot create claims.
++
+.. Create a new project request template from the modified `template.yaml` file in the `openshift-config` namespace:
++
+----
+$ oc create -f template.yaml -n openshift-config
+----
++
+[NOTE]
+====
+To include the configuration as a `kubectl.kubernetes.io/last-applied-configuration` annotation, add the `--save-config` option to the `oc create` command.
+====
++
+By default, the template is called `project-request`.
++
+* If a project request template already exists within a cluster:
++
+.. List templates in the `openshift-config` namespace:
++
+----
+$ oc get templates -n openshift-config
+----
++
+.. Edit an existing project request template:
++
+----
+$ oc edit template <project_request_template> -n openshift-config
+----
++
+[NOTE]
```

Comment:
It sounds like you should see this note before you perform the steps, so please move it to the start of the substep.

Reply:
Resolved.